ui/app/components/app/account-menu/account-menu.component.js

```diff
@@ -37,35 +41,96 @@ export default class AccountMenu extends PureComponent {
     toggleAccountMenu: PropTypes.func,
   }
 
+  accountsRef;
+
   state = {
-    atAccountListBottom: false,
+    shouldShowScrollButton: false,
+    searchQuery: '',
   }
 
-  componentDidUpdate (prevProps) {
-    const { prevIsAccountMenuOpen } = prevProps
+  addressFuse = new Fuse([], {
+    shouldSort: false,
+    threshold: 0.45,
+    location: 0,
+    distance: 100,
+    maxPatternLength: 32,
+    minMatchCharLength: 1,
+    keys: [
+      { name: 'name', weight: 0.5 },
+      { name: 'address', weight: 0.5 },
+    ],
+  })
+
+  componentDidUpdate (prevProps, prevState) {
+    const { isAccountMenuOpen: prevIsAccountMenuOpen } = prevProps
+    const { searchQuery: prevSearchQuery } = prevState
     const { isAccountMenuOpen } = this.props
+    const { searchQuery } = this.state
 
     if (!prevIsAccountMenuOpen && isAccountMenuOpen) {
-      this.setAtAccountListBottom()
+      this.setShouldShowScrollButton()
+      this.resetSearchQuery()
+    }
+
+    // recalculate on each search query change
+    // whether we can show scroll down button
+    if (isAccountMenuOpen && prevSearchQuery !== searchQuery) {
+      this.setShouldShowScrollButton()
     }
   }
 
+  renderAccountsSearch () {
+    const inputAdornment = (
+      <InputAdornment
+        position="start"
+        style={{
```

Comment:
@Gudahtt this one I'm really not happy with but seems for some reason material styles are with higher priority (that's probably a bug with the order how styles are imported) than `account-menu` styles. So either I can use the custom class with `!important`s or inline styles. Let me know if you see some other option here that's more appropriate